src/main/java/io/cdap/plugin/cdc/source/salesforce/SalesforceReceiver.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.cdc.source.salesforce;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.sforce.soap.partner.PartnerConnection;
+import com.sforce.soap.partner.QueryResult;
+import com.sforce.soap.partner.sobject.SObject;
+import com.sforce.ws.ConnectionException;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.cdc.common.ErrorHandling;
+import io.cdap.plugin.cdc.common.OperationType;
+import io.cdap.plugin.cdc.source.salesforce.authenticator.AuthenticatorCredentials;
+import io.cdap.plugin.cdc.source.salesforce.records.ChangeEventRecord;
+import io.cdap.plugin.cdc.source.salesforce.records.SalesforceRecord;
+import io.cdap.plugin.cdc.source.salesforce.sobject.SObjectDescriptor;
+import io.cdap.plugin.cdc.source.salesforce.sobject.SObjectsDescribeResult;
+import io.cdap.plugin.cdc.source.salesforce.util.SalesforceConnectionUtil;
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.receiver.Receiver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of Spark receiver to receive Salesforce change events
+ */
```

Comment:
please add more javadocs about what this class is doing. More specifically, how it consumes the change events and processes them.

Reply:
There needs to be far more documentation here. What are the different types of events that can be read from Salesforce? What are the types of events that this emits? Examples?